transport/graphsync/extension/gsextension.go

```diff
@@ -65,14 +71,22 @@ type GsExtended interface {
 //    * nil + error if the extendedData fails to unmarshal
 //    * unmarshaled ExtensionDataTransferData + nil if all goes well
 func GetTransferData(extendedData GsExtended) (datatransfer.Message, error) {
-	extName := ExtensionDataTransfer1_1
+	extNames := []graphsync.ExtensionName{
+		ExtensionIncomingRequest1_1,
+		ExtensionOutgoingBlock1_1,
+		ExtensionDataTransfer1_1,
+		ExtensionDataTransfer1_0,
+	}
+	i := 0
+	extName := extNames[i]
 	data, ok := extendedData.Extension(extName)
-	if !ok {
-		extName = ExtensionDataTransfer1_0
-		data, ok = extendedData.Extension(extName)
-		if !ok {
+	for !ok {
+		i++
+		if i == len(extNames) {
```

Comment:
One more nit - can we restructure this for loop a little, eg:
```
var ok bool
var []byte data
for i := 0; i < len(extNames) && !ok; i++ {
  data, ok = extendedData.Extension(extNames[i])
}
if !ok {
  return nil, nil
}
```

Reply:
Good call thanks!

Reply:
Looks good 👍